core/Columns/Dimension.php

```diff
@@ -758,13 +758,35 @@ public static function getDimensions(Plugin $plugin)
         $columns = $plugin->findMultipleComponents('Columns', '\\Piwik\\Columns\\Dimension');
         $instances  = array();
 
-        foreach ($columns as $colum) {
-            $instances[] = new $colum();
+        foreach ($columns as $column) {
+            $instances[] = new $column();
         }
 
+        /**
+         * Triggered to filter / restrict dimensions.
+         *
+         * @param Dimension[] $dimensions An array of dimensions
+         */
+        Piwik::postEvent('Dimension.filterDimensions', array(&$instances));
```

Comment:
@sgiehl There's already a `Dimension.filterDimension` further above. Will debugging noticed it might not be needed to have it here as well and it's enough in `getDimensions()`? If we keep it here then we should in  `removeOldDimensions ` check if `$instances` is empty and return quickly not to get the removed dimensions 100 times when most plugins don't define any. I haven't fully tested but maybe it could be even removed the event here?

Not that it's a big change but I'm thinking we could even have the check for removed dimensions directly in this class for simplicity and wouldn't need an event for it.

Reply:
Actually we need both calls of `Dimension.filterDimensions`. If someone directly uses the `getDimensions` method the dimensions actually weren't filtered before. Which was imho no correct behaviour. Only doing it in `getDimensions` would make it impossible to filter out a dimension that was added by another plugin using `Dimension.addDimensions` event.

Removing such dimensions here directly sound fine as well. Will move it here

Reply:
done.

Reply:
@sgiehl ok. Can we remove the event below though? That would be great. It be fine for now if not all dimensions are 100% filtered in these cases I would say since most of the time "all dimensions" would be called. I'd want to avoid potentially triggering a lot of events here slowing Matomo down in some cases.